inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -569,7 +569,7 @@ protected function get_sql_where_clause( $post_type ) {
 		// Based on WP_Query->get_posts(). R.
 		if ( 'attachment' === $post_type ) {
 			$join   = " LEFT JOIN {$wpdb->posts} AS p2 ON ({$wpdb->posts}.post_parent = p2.ID) ";
-			$status = "p2.post_status = 'publish'";
+			$status = "p2.post_status = 'publish' AND p2.post_password = ''";
```

Comment:
I thought at first that `p2.post_password > ''` would be preferable, as it covers both an empty string and the NULL value, but WordPress does not have any nullable columns, of course...

LGTM.